### ISSUE TITLE
Probabilistic assignment causes considerable memory allocation

### ISSUE THREAD

**Issue by dickscheid** [FZJ-INM1-BDA/siibra-python#139]
Reported by Andrea Brovelli: probabilistic assignment with the Julich-Brain maps requires more then 2GB of main memory.

This is a hint to a flaw in some caching behaviour in image loading, since the ParcellationMap class itself is already designed to walk map by map and avoid allocation of multiple maps in parallel. Probably even a bug.

**Comment by dickscheid** [comment]
> related to #137 ?

I already tracked the problem down for this one, will then check if it solves #137 as well. Not related to memoization, it is a flaw in the design of LazyHttpLoader. A patch will be pushed soon.

**Comment by dickscheid** [comment]
This has now been adressed in cdfa626798c97f71d9f38a6699fb41c4cacd8716, after some more testing we will release a new package.

**Comment by dickscheid** [comment]
Fixed by v0.3a5

**Comment by dickscheid** [comment]
Just to add a bit more explanation here: The memory consumption was caused by the design flaw in LazyHttpLoader, however, fixing that flaw was making probabilitstic assignment very slow. Therefore, a reimplementation of the ContinuousParcellationMap was performed.
